operators/src/pro/cache/shared_cache.rs

```diff
@@ -0,0 +1,1190 @@
+    /// It will add the element to the landing zone entry of the query.
+    ///
+    /// # Errors
+    ///
+    /// This method returns an error if the query is not in the landing zone.
+    /// This method returns an error if the element is already expired.
+    /// This method returns an error if the landing zone is full or the new element would cause the landing zone to overflow.
+    ///
+    fn add_query_element_to_landing_zone(
+        &mut self,
+        query_id: &QueryId,
+        landing_zone_element: C,
+    ) -> Result<(), CacheError> {
+        let landing_zone_entry = self
+            .operator_cache
+            .landing_zone_entry_mut(query_id)
+            .ok_or(CacheError::QueryNotFoundInLandingZone)?;
+
+        if landing_zone_element.cache_hint().is_expired() {
+            log::trace!("Element is already expired");
+            return Err(CacheError::TileExpiredBeforeInsertion);
+        };
+
+        let element_bytes_size = landing_zone_element.byte_size();
+
+        if !self.landing_zone_size.can_fit_bytes(element_bytes_size) {
+            return Err(CacheError::NotEnoughSpaceInLandingZone);
+        }
+
+        // new entries might update the query bounds stored for this entry
+        landing_zone_element.update_stored_query(&mut landing_zone_entry.query)?;
+
+        // acctually insert the element into the landing zone
+        landing_zone_entry.insert_element(landing_zone_element)?;
+
+        // we add the bytes size of the element to the landing zone size after we have inserted it.
+        self.landing_zone_size
+            .try_add_bytes(element_bytes_size)
+            .expect(
+            "The Landing Zone must have enough space for the element since we checked it before",
+        );
+
+        log::trace!(
+            "Inserted tile for query {} into landing zone. Landing zone size: {}. Landing zone size used: {}. Landing zone used percentage: {}",
+            query_id, self.landing_zone_size.total_byte_size(), self.landing_zone_size.byte_size_used(), self.landing_zone_size.size_used_fraction()
+        );
+
+        Ok(())
+    }
+
+    /// This method inserts a query into the landing zone.
+    /// It will cause the operator cache to create a new landing zone entry.
+    /// Therefore, the size of the query and the size of the landing zone entry will be added to the landing zone size.
+    ///
+    /// # Errors
+    ///
+    /// This method returns an error if the query is already in the landing zone.
+    /// This method returns an error if the landing zone is full or the new query would cause the landing zone to overflow.
+    ///
+    fn insert_query_into_landing_zone(&mut self, query: &C::Query) -> Result<QueryId, CacheError> {
+        let landing_zone_entry = CacheQueryEntry::create_empty::<C>(query.clone());
+        let query_id = QueryId::new();
+
+        self.landing_zone_size.try_add_element_bytes(&query_id)?;
+        self.landing_zone_size
```

Comment:
don't you have to subtract the query_id byte size again if this fails?

Reply:
7edc0303f509e4569e10c4e0011cc67530d03581

Reply:
[7edc030](https://github.com/geo-engine/geoengine/pull/833/commits/7edc0303f509e4569e10c4e0011cc67530d03581)